### DISCUSSION TITLE
Boolean AND GeometricObjects

### DISCUSSION THREAD

**Discussion by dalarev** [NanoComp/meep#2660] [Q&A]
Is there a way to Boolean AND `GeometricObject`s?

I'm trying to generate a biconvex lens via the AND of two `Cylinder` objects, `C1` and `C2`. `C1 and C2` won't work.

Appreciate the help!

![2Dlens](https://github.com/NanoComp/meep/assets/52722332/bceafba2-1d30-4068-bc39-3749272e9fa3)

**Comment by dalarev** [comment]
Abandoned this in favor of the `default_geometry` method [(link to example)](https://github.com/NanoComp/meep/discussions/2662).

**Comment by stevengj** [comment]
libctl doesn't currently support taking the intersection of two geometric objects.   You can, of course, define your own geometry manually (by importing from a numpy array, using `epsilon_input_file`, or the `material_function` feature.

(The only drawback of these methods is that they will not do subpixel smoothing for you at interfaces, unlike geometric objects.)
